addon/types.ts

```diff
@@ -0,0 +1,36 @@
+import Mix from '@ember/polyfills/types';
+
+export type Nullable<T> = T | null | undefined;
+
+export interface PlainObject {
+  [key: string]: string | PlainObject | PlainObject[];
+}
+
+export interface PlainHeaders {
+  [key: string]: string | undefined | null;
```

Comment:
```suggestion
  [key: string]?: string | null;
```

Reply:
I think typescript doesn't let me do this.

Reply:
you'll have to change it to a `type` instead of `interface`

```ts
interface Foo{
    bar: string;
    baz: number;
}

type A<T> = { [K in keyof T]?: T[K] | null }

type X = A<Foo>;/*
{
    bar?: string | null | undefined;
    baz?: number | null | undefined;
}
*/

```
https://typescript-play.js.org/#code/JYOwLgpgTgZghgYwgAgGIHt0G8BQz-IBGcUAXMgM5hSgDmA3HgcQF7kgCuAtodIwL44cYAJ4AHFAEEAPABUAfMgC8yLMgDaAaWShkAawgj0MZLIC6AfnKytZ5AB9knADbPkg4eJQANZchkY6PL0APQAVDi4BEQkVpTUdA5OHK5JHCAAJhAwoBAZjNGscZw80Ekubo7pWTkgeQI4YSE4QA

Reply:
I changed `PlainHeaders` and `PlainObject` from `interface` to `type`, and made `PlainObject` more generic.